src/main/java/mekanism/common/registries/MekanismModules.java

```diff
@@ -54,19 +55,17 @@ private MekanismModules() {
           ModuleAttackAmplificationUnit::new, () -> MekanismItems.MODULE_ATTACK_AMPLIFICATION.asItem(), builder -> builder.maxStackSize(4).rarity(Rarity.UNCOMMON)
                 .rendersHUD());
     public static final ModuleRegistryObject<ModuleFarmingUnit> FARMING_UNIT = MODULES.register("farming_unit", ModuleFarmingUnit::new,
-          () -> MekanismItems.MODULE_FARMING.asItem(), builder -> builder.maxStackSize(4).rarity(Rarity.UNCOMMON).exclusive());
-    //TODO: Eventually we may want to come up with a better exclusive method given realistically the shearing unit and farming unit don't need to be
-    // exclusive of each other, but they both should be exclusive in regards to the teleportation unit
+          () -> MekanismItems.MODULE_FARMING.asItem(), builder -> builder.maxStackSize(4).rarity(Rarity.UNCOMMON).exclusive(ExclusiveFlag.getFlags(ExclusiveFlag.INTERACT_BLOCK)));
     public static final ModuleRegistryObject<ModuleShearingUnit> SHEARING_UNIT = MODULES.register("shearing_unit", ModuleShearingUnit::new,
-          () -> MekanismItems.MODULE_SHEARING.asItem(), builder -> builder.rarity(Rarity.UNCOMMON).exclusive());
+          () -> MekanismItems.MODULE_SHEARING.asItem(), builder -> builder.rarity(Rarity.UNCOMMON).exclusive(ExclusiveFlag.getFlags(ExclusiveFlag.INTERACT_ENTITY)));
```

Comment:
Might be worth a comment that while you can shear blocks in general they don't overlap with anything that supports the farming unit right click.

Reply:
With that in mind though, do we want to split the INTERACT_BLOCK flag to clarify that?

Reply:
I mean we can just split Interact block to shearable/not and let addons deal with it. It would kinda be on them if they made the farming module do something to a shearable block if we set its exclusivity to say it doesn't. Also they could potentially override its mask to say it does.

Somewhat related, what do you think about trying to deal with dynamic exclusivity. For example the teleport module could technically just be INTERACT_BLOCK if the requires block config is active. I think it's doable, we'd just have to check exclusivity with most of the config changes and they'd be defining a mask supplier instead of a constant mask.

Reply:
I feel like dynamic exclusivity isn't worth the headache of taking into account then disabling other conflicting modules when other sub settings are changed. And I think just make the shear one for now also have INTERACT_BLOCK given some mod may add a block that can be both sheared and tilled, hoed, etc